xrtmodels/xrtconv.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2022 IOTech Ltd
+
+package xrtmodels
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+)
+
+func toXrtProperties(protocol string, protocolProperties map[string]interface{}) errors.EdgeX {
+	propertyToInt, propertyToFloat := propertyConversionList(protocol)
+
+	for _, p := range propertyToInt {
+		propertyValue, ok := protocolProperties[p]
+		if ok {
+			// convert property value from interface{} to string, then to int
+			val, err := strconv.Atoi(fmt.Sprintf("%v", propertyValue))
+			if err != nil {
+				return errors.NewCommonEdgeX(errors.KindContractInvalid, fmt.Sprintf("fail to convert %v to int", p), err)
+			}
+			protocolProperties[p] = int(val)
+		}
+	}
+
+	for _, p := range propertyToFloat {
+		propertyValue, ok := protocolProperties[p]
+		if ok {
+			// convert property value from interface{} to string, then to float
+			val, err := strconv.ParseFloat(fmt.Sprintf("%v", propertyValue), 64)
+			if err != nil {
+				return errors.NewCommonEdgeX(errors.KindContractInvalid, fmt.Sprintf("fail to convert %v to float", p), err)
+			}
+			protocolProperties[p] = val
+		}
+	}
+	return nil
+}
+
+func toEdgeXProperties(protocol string, protocolProperties map[string]interface{}) map[string]string {
+	propertyToInt, propertyToFloat := propertyConversionList(protocol)
```

Comment:
rename variable to `intProperties` and `floatProperties`

Reply:
Renamed.